lib/sidekiq/fetch.rb

```diff
@@ -27,14 +27,11 @@ def requeue
     def initialize(options)
       @strictly_ordered_queues = !!options[:strict]
       @queues = options[:queues].map { |q| "queue:#{q}" }
-      if @strictly_ordered_queues
-        @queues.uniq!
-        @queues << TIMEOUT
-      end
+      @queues.uniq! if @strictly_ordered_queues
     end
 
     def retrieve_work
-      work = Sidekiq.redis { |conn| conn.brpop(*queues_cmd) }
```

Comment:
This does have the effect of allocating a hash.  Is it better to use a possibly deprecated API and save a hash allocation or not?

Reply:
Thanks, Mike,

I ran some benchmarks using `benchmark-ips` and the new method call is around 3x slower than the current method, so, purely if the decision is purely performance-based - probably not a good idea to merge.

But if the deprecation happens at some point, we'd be forced to use this anyway. Given the fact that the deprecation does not have a fixed date (it has been that way for at least a couple of years now), I do not have a strong opinion, to be honest 🙂